src/Microsoft.Azure.WebJobs.Host/Bindings/BindingProviders/ExactTypeBindingProvider.cs

```diff
@@ -96,7 +102,14 @@ private class ExactBinding : BindingBase<TAttribute>
 
                 IValueProvider vp = new ConstantValueProvider(obj, typeof(TUserType), invokeString);
                 return vp;
-            }            
+            }
+
+            protected override Task<IValueProvider> BuildAsync(TAttribute attrResolved, BindingContext context)
+            {
+                _preBindHook?.Invoke(attrResolved, context);
```

Comment:
Doesn't resolving parameters like `{id}` happen at invoke time? You're doing it inside the pre-bind hook right now. I'm trying to think through the pattern for people that want to do custom binding and not use `[AutoResolve]` -- which we'll need to do for the DocDB SqlQuery feature as well.

You brought it up elsewhere already -- you want to construct the template during index time, which explains the need for a 'post-resolve hook'. But I'm curious whether you'd need a 'pre-bind hook' if we just passed the BindingContext directly to the 'rule' delegate.